storage-access.bs

```diff
@@ -224,9 +234,41 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>re
 
 ISSUE: Shouldn't step 3.7 be [=same site=]?
 
+When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>requestStorageAccessForSite(site)</code></dfn> method must run these steps:
+
+1. Let |p| be [=a new promise=].
+1. If this algorithm was invoked when |doc|'s {{Window}} object did not have [=transient activation=], [=reject=] and return |p|.
+1. If |doc|'s [=Document/browsing context=] is not a [=top-level browsing context=], [=reject=] and return |p|.
+1. If |doc|'s [=Document/origin=] is an [=opaque origin=], [=reject=] and return |p|.
+1. Let |parsedURL| be the the result of running the [=URL parser=] on |site|.
+1. If |parsedURL| is failure, [=reject=] and return |p|.
+1. Let |origin| be |parsedURL|'s [=url/origin=].
+1. If |origin| is an [=opaque origin=], [=reject=] and return |p|.
+1. If |doc|'s [=Document/origin=] is [=same origin=] with |origin|, [=/resolve=] and return |p|.
+<!-- TODO: generate a partitioned storage key assumes (top-level site, site) come from |doc| and doesn't have the possibility of override. -->
+<!-- we can either change those steps to take two site parameters or write new steps -->
+<!-- also could pass |parsedURL| instead of |origin|, or could pass |site| directly-->
+1. Let |key| be the result of [=generate an overridden partitioned storage key|generating an overridden partitioned storage key=] from |doc| and |origin|.
+1. If |key| is failure, [=reject=] and return |p|.
+1. Let |global| be |doc|'s [=relevant global object=].
+<!-- TODO: may need to make 'obtain the storage access map' generic too, or otherwise indicate that |doc| in the other function would be like "as if there was a |doc| whose site was the passed-in site" -->
```

Comment:
Sorry, can you elaborate on what the issue is? :)

Reply:
I think I was concerned about race conditions with the way the storage access map is copied per document, where you could conceivably have more than one copy and simultaneous requests to `requestStorageAccess` and `requestStorageAccessForSite`, but I think this is irrelevant...I removed it